accepted/2024/improve-support-for-packages-signed-with-trusted-signing.md

```diff
@@ -0,0 +1,155 @@
+is now linked to the user. That will mean that the user would need to contact the NuGet team to remove the EKU link until updates
+to the interface have been made. But I don't think a lot of people will need this functionality right away.
+
+The EKU will only be linked when the following conditions are true:
+- The package was signed within the last X days where X should be decided by the NuGet team (not sure if this is really necessary).
+- The signing certificate is signed by `Microsoft Identity Verification Root Certificate Authority 2020`.
+  (https://learn.microsoft.com/en-us/azure/trusted-signing/concept-trusted-signing-trust-models)
+- The signing certificate has a valid counter-signature (timestamp).
+- The signing certificate contains the Trusted Signing EKU.
+- The signing certificate contains a Public Trust Identity EKU.
+- The Public Trust Identity EKU is not already linked to the user.
+
+This mean that the following would happen inside the service that handles the upload of NuGet packages:
+
+![NuGet package upload](images/trusted-signing-upload.png)
+
+As shown in the image there are is no check if the certificate is expired when the user uploaded a Trusted Signing certificate.
+This now happens when the user uploads a certificate but this check should not be added to Trusted Signing certificates because
+they will expire after three days.
+
+#### How does this help the user?
+
+This helps the user because they will now no longer need to upload a `.cer` file everytime they publish a new NuGet package that
+was signed with the same Trusted Signing certificate. But inside the NuGet Gallery they will no longer see the certificate that
+they uploaded earlier. This could confuse the user so we could keep the link and let the user remove it manually. But doing this
+would also not make it clear that an EKU was linked to their account. Making this visible in the interface will be the next step
+of these changes.
+
+### Step 2: Change how Trusted Signing EKU's are shown in the NuGet Gallery.
+
+#### How would this be implemented?
+
+The next steps would be to change how this will be displayed inside the NuGet Gallery. Below is an image of how this could be
+added to the account page of a user (please forgive me for my css skills). The naming is a proposal and I am open to suggestions
+but I think it would be best to show the EKU's in a separate table.
+
+![User profile](images/trusted-signing-profile.png)
+
+This image demonstrates how this would look when a user has both an EKU and certificate linked to their account. As mentioned in
+step 1 the certificate would no longer be shown and the user would only see their EKU's. The Subject and Issuer will be extracted
+from the certificate the first time and added to the EKU data. Uploads in the future could result in a new Issuer so this should
+be updated but a change of Subject would result in a different EKU so there is no need to update this.
```

Comment:
With Trusted Signing, the change in subjectDN will not result in a different EKU if the organization/individual legally changes their name and validates that new name with their existing Trusted Signing resources. If they choose to use a new subscription and identity validation for the new legal name, a new unique EKU would result.

Reply:
Thanks for mentioning this. Seems that I misunderstood you earlier. I have updated the PR.